src/components/pages/TeacherDashboard/TeacherViewRumble/RumbleStudentList/RumbleStudentListContainer.tsx

```diff
@@ -10,27 +11,23 @@ const RumbleStudentListContainer = ({
   const [studentList, setStudentList] = useState<
     Students.IStudentWithSubmissions[]
   >();
-  const [error, setError] = useState<string>();
 
+  const [getWithSubsByRumbleId, loading, , error] = useAsync({
+    asyncFunction: Students.getWithSubsByRumbleId,
+    setter: setStudentList,
+  });
   useEffect(() => {
-    console.log('rumbleId', rumble.id);
-    setStudentList(undefined);
-    Students.getWithSubsByRumbleId(rumble.id)
-      .then((res) => {
-        setStudentList(res);
-      })
-      .catch((err) => {
-        console.log(err);
-        setError(err.message);
-      });
+    getWithSubsByRumbleId(rumble.id);
   }, [section, rumble]);
 
   return studentList && section ? (
     <RenderRumbleStudentList studentList={studentList} section={section} />
   ) : error ? (
-    <CouldNotLoad error={error} />
-  ) : (
+    <CouldNotLoad error={error.message} />
+  ) : loading ? (
     <Loader message={'Loading students'} />
+  ) : (
```

Comment:
We should have an error or a list at this point, we don't actually need to check the loading state here.